crypto/blake2/blake2b_mac.c

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright 2018 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifndef OPENSSL_NO_BLAKE2
+
+# include <openssl/evp.h>
+# include "blake2_locl.h"
+# include "internal/cryptlib.h"
+# include "internal/evp_int.h"
+
+/* typedef EVP_MAC_IMPL */
+struct evp_mac_impl_st {
+    BLAKE2B_CTX *S;
```

Comment:
Since you have included `blake2_locl.h`, I don't see a point to having a pointer here.  `BLAKE2B_CTX S;` saves you the need to perform yet another allocation.

It would of course be a different thing if `BLAKE2B_CTX` was opaque, but then you would also need a `BLAKE2B_New()` to allocate it, and well, you don't...